servant-util/src/Servant/Util/Combinators/Filtering/Base.hs

```diff
@@ -90,6 +97,18 @@ unsupportedFilteringValue errMsg = FilteringValueParser (\_ -> Left errMsg)
 -- This is not a 'Map' to prevent developer-defined entries order.
 type OpsDescriptions = [(Text, Text)]
 
+-- | Specify the encoding type and function for query parameters.
+-- It's required due to this https://github.com/haskell-servant/servant/pull/1432
+#if MIN_VERSION_servant(0,19,0)
+type EncodedQueryParam = ByteString
+encodeQueryParam :: ToHttpApiData a => a  -> EncodedQueryParam
+encodeQueryParam = BL.toStrict . toLazyByteString . toEncodedUrlPiece
+#else
+type EncodedQueryParam = Text
+encodeQueryParam :: ToHttpApiData a => a  -> EncodedQueryParam
+encodeQueryParam = toQueryParam
+#endif
```

Comment:
This seems really awkward. How bad would it be to bump the dependency version?

Reply:
I do not like this solution too. If supporting servant 0.18 is not required, let's remove it.
Btw, I'll create an other PR to export `encodeQueryParam` from the `servant-client-core` package.

Reply:
Did you were thinking something like this https://github.com/marinelli/servant-util/commit/a0dbeeb24212e1a21e0fc332abc5523e5d939029?

Reply:
@Martoon-00 what do you think about it?

Reply:
I apologize for the late reply :pray: :pray: 

I don't like such tricks with CPP pragmas too, but in this case its scope is sufficiently limited and it seems like the least evil to me.

Let's go with the current solution to leave others some time if they need a recent feature of `servant-util` but cannot afford to switch to the `servant-0.19`. And after a couple of major releases of `servant`, or if this `CPP` turns out to hinder other features, we will strip the old behaviour.

I also asked our OPS team to make CI run both the old and new versions of `servant-client-core` ([internal ticket for this](https://issues.serokell.io/issue/OPS-1315)).

---

@marinelli You are also very welcome to start using https://github.com/haskell-servant/servant/pull/1549 here as soon as it appears in a release. Just leave a note in the changelog that we do not support `servant-0.19`, rather only `< 0.19` and `>= 0.19.X`.

Thanks a lot for your work!

Reply:
I'm not sure anymore that re-exporting `encodeQueryParamValue` from `servant-client-core` is a good idea :)
We'll see.